src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-sysfs.c

```diff
@@ -86,8 +86,12 @@ static ssize_t mfg_ver_show(struct device *dev,
 	struct device_attribute *attr, char *buf)
 {
 	struct xclmgmt_dev *lro = dev_get_drvdata(dev);
+	struct VmrStatus vmr_header = {};
 
-	return sprintf(buf, "%d\n", MGMT_READ_REG32(lro, _GOLDEN_VER));
+	/*Checks for Versal VMR availability; returns -EINVALID for Versal VMR based device.*/
```

Comment:
what about:
/*
 * For non-versal platform, ret is -ENODEV, we can proceed and read register from PCIE bar.
 * For versal platform, no matter ret is 0 or any error, we should return EINVAL.
 * /